### PR TITLE
feat: Create the update script for 7.4 to 7.5

### PR DESCRIPTION
**Pull request by liordino**
feat: update node to version 14

feat: update supervisor config

fix: removed check for version 7.5 install

fix: install curl prior to updating node

fix: make everything unattended

fix: update to ubuntu 20.04

fix: divided the script in two parts, one to upgrade ubuntu to 18.04 and
another do update it to 20.04 and do the all the rest

### CODE REVIEW

**Review by chaotixkilla — APPROVED** [review]

File: upgrade [hunk 1]
```diff
@@ -30,82 +28,110 @@ if [ "$(whoami)" != "root" ]; then
...
+  chmod -x /etc/update-motd.d/*-motd-news 2>/dev/null
+  set -e
+
+  # reset terminal to link new executables
+  hash -r
+
+  rm -r /var/lock/lamassu-update
+
+  echo "Update part 3 complete! Restart the server to complete the process."
+elif [[ "$UBUNTU_VERSION" == "20.04" ]]; then
```
Comment (naconner) [inline upgrade:133]: This doesn't mean the system is up-to-date. This same script needs to accomodate operators who had installed v7.5.0 fresh for the first time, starting with 20.04, and who in the future need to go from v7.5.0 to v7.5.1, or whatever the latest version is.

**Review by CrypticaScriptura — CHANGES_REQUESTED** [review]
This script needs to update operators currently on v7.5 to any future l-s versions, as it's currently meant to. Whereas these changes are only good for updating v7.4 servers once.

File: upgrade [hunk 2]
```diff
@@ -30,82 +28,110 @@ if [ "$(whoami)" != "root" ]; then
...
+  sudo n 14
+  sudo n which 14
+
+  decho "updating to ubuntu 18.04"
+  sudo apt-get update -y
+  sudo apt-get --with-new-pkgs upgrade -y
+  sudo apt-get dist-upgrade -y
+  sudo do-release-upgrade -f DistUpgradeViewNonInteractive
+
+  echo "Update part 1 complete! Reboot the server, make sure that Ubuntu is now on version 18.04 and run this script again to execute the second step."
```
Comment (naconner) [inline upgrade:52]: Part 1 of how many? How does an operator reboot the server? How does one make sure it's on 18.04 (and doesn't the script check that for them)?

Also, use spacing:

```
echo
echo "Update part 1 of 2 complete!"
echo "You must now reboot this server, log in again, and run this script again to complete the upgrade."
echo "Paired machines will not be online until you do so."
echo "Reboot your server by running the following command:"
echo
echo "shutdown -r now"
echo
```

File: upgrade [hunk 3]
```diff
@@ -30,82 +28,110 @@ if [ "$(whoami)" != "root" ]; then
...
+elif [[ "$UBUNTU_VERSION" == "18.04" ]]; then
+  echo "Detected Ubuntu version: 18.04. Executing upgrade script part 2 (you may need to reboot more than once to finish this step)..."
```
Comment (naconner) [inline upgrade:54]: Reboot more than once? How do they know if and when to?

File: upgrade [hunk 4]
```diff
@@ -30,82 +28,110 @@ if [ "$(whoami)" != "root" ]; then
...
+  chmod -x /etc/update-motd.d/*-cloudguest 2>/dev/null
+  chmod -x /etc/update-motd.d/*-motd-news 2>/dev/null
+  set -e
+
+  # reset terminal to link new executables
+  hash -r
+
+  rm -r /var/lock/lamassu-update
+
+  echo "Update part 3 complete! Restart the server to complete the process."
```
Comment (naconner) [inline upgrade:132]: Hadn't informed somewhere that we were going from part 2 to a part 3.

```
echo
echo "Update part 2 of 2 complete!"
echo "You must now reboot this server to finish the update."
echo "Paired machines will not be online until you do so."
echo "Reboot your server by running the following command:"
echo
echo "shutdown -r now"
echo
```

**Review by CrypticaScriptura — CHANGES_REQUESTED** [review]
Need to instruct operators clearly.

**Pull request closed without merging**